docs/integrations/ai-engines/llamaindex.mdx

```diff
@@ -0,0 +1,86 @@
+## LlamaIndex
+
+LlamaIndex is a data framework for your LLM application that allows you to work with private or domain-specific data and simplifies the interface between your data source and LLM.
+
+Developers can use LlamaIndex to build chatbots, applications, private set-ups, and Q&A over documents and web pages. The implemented features for LlamaIndex is [Support Web Page Reader](https://gpt-index.readthedocs.io/en/latest/examples/data_connectors/WebPageDemo.html) and [Support Database Reader](https://gpt-index.readthedocs.io/en/latest/examples/data_connectors/DatabaseReaderDemo.html)
+
+For this use case, you will be creating a question-answering model that retrieves answers from a webpage. OpenAI will be used as the LLM and Blackrock's investment webpage will be used as the datasource.
+
+### How to bring LlamaIndex to MindsDB
+
+The `CREATE ML_ENGINE` syntax is used to create the engine for LlamaIndex.
+
+```sql
+CREATE ML_ENGINE llamaindex
+FROM llama_index
+```
+
+On execution you get:
+<img src="/assets/tutorials/llamaindex/1.ml_engine.png" />
+
+Use the `CREATE MODEL` syntax to create a model with Llamaindex.
+
+```sql
+CREATE MODEL qa_blackrock
+FROM files
+    (SELECT * FROM about_blackrock)
+PREDICT Answers
+USING 
+  engine = 'llamaindex', 
+  index_class = 'GPTVectorStoreIndex',
+  reader = 'DFReader',
+  source_url_link = 'https://www.blackrock.com/za/individual/about-us',
+  input_column = 'Questions',
+  openai_api_key = 'your_api_key';
+  ```
+
+Where:
+
+| Expression                             | Description                                |
+|----------------------------------------|--------------------------------------------| 
+| `qa_blackrock`                         | The name provided to the model.            |
+| `engine`                               | The engine used,which is Llamaindex.       |
+| `index_class`                          | The vector store index used.               |
+| `reader`                               |The reader that reads the data/webpage.     |
```

Comment:
Please add `source_url_link` after `reader`. And description: `The webpage link.`